qiskit/compiler/transpiler.py

```diff
@@ -561,128 +319,144 @@ def _parse_transpile_args(
     # If target is not specified and any hardware constraint object is
     # manually specified then do not use the target from the backend as
     # it is invalidated by a custom basis gate list or a custom coupling map
-    elif basis_gates is None and coupling_map is None:
-        target = _parse_target(backend, target)
+    elif basis_gates or coupling_map:
+        _skip_target = True
```

Comment:
Minor: this probably wants to maintain the `is None` checks, because `basis_gates=[]` is an explicit (pathological) setting of `basis_gates` that should probably be respected.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/10291/commits/0ea2edae911f6b9f02bc2dc627a0896853f4efc1 although my guess is the functional difference between `basis_gates = []` and `basis_gates = None` is the empty list will error 100% of the time in basis translation.